### PR TITLE
implement support for sub-menues (bsc #906990)

### PR DESCRIPTION
**Pull request by wfeldt**
See test/menu/isolinux.cfg for a sample config with two nested sub-menues.

### CODE REVIEW

**Comment by wfeldt** [comment]
There's a sample iso to try at ~snwint/Export/menu.iso.

File: themes/openSUSE/src/submenu.inc [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+% Menu entries that have an 'append' entry '->BAR' link to submenu BAR.
+%
+% As there are spaces in labels required to make this work, use the 'menu
+% label' statement in syslinux.
+
+% submenu entries must start with this string, followed by an arbitrary identifier
+/submenu.tag "->" def
+
+% submenues are stacked here
+/submenu.list 8 array def
```
Comment (jreidinger) [inline themes/openSUSE/src/submenu.inc]: does it mean, that number of submenu elements are limited to 8? if so, is it documented somewhere?

**Comment by jreidinger** [comment]
Note related to question about limitation. Where it is documented? ( beside comment above code? ) I think it make sense to have some documentation how cfg look like and what is its limitation.

File: themes/openSUSE/src/submenu.inc (hunk 1 above)
Comment (wfeldt) [inline themes/openSUSE/src/submenu.inc]: The nesting level is 8; it will just refuse to enter deeper submenues. I think this is good enough...

Comment (jreidinger) [inline themes/openSUSE/src/submenu.inc]: I am also fine with such limit, I just think we should document it somewhere else, not just burden it inside code.

**Comment by jreidinger** [comment]
thanks for documentation. LGTM

**Pull request merged**